WordPress/gutenberg
# Image: Add border block support for color, width, and style

Fixes: #38801

Depends on: 
- https://github.com/WordPress/gutenberg/pull/42087

Part of: 
- https://github.com/WordPress/gutenberg/issues/21540

## Description

This takes advantage of border block support to add those features to the Image block.

- Individual image border support added via block supports (skips serialization and manually applies to the inner img)
- Leverages block support feature level selectors added in [#42087](https://github.com/WordPress/gutenberg/pull/42087) to allow theme.json and Global Styles theming of borders.
- Theme.json can style the image borders via normal without having to use the additional nesting of the elements API

## How has this been tested?
Fixture tests and manually.

#### Note: There is a current bug on `trunk` impacting resetting of controls. 

#### Test Instructions
1. Create a post, add an image block, set a border-radius on the image block and save the post
2. Checkout this PR and reload the editor
3. Ensure there are no block validation errors and the block still displays its border-radius correctly
4. Set new border color, radius and width on the image block ensuring it updates correctly
5. From the block controls toolbar, choose to crop the image
6. Ensure the border still displays
7. Cancel the cropping and save the post
8. Confirm the image border displays correctly on the frontend
9. Back in the editor add a gallery, with images. Apply different borders and captions to the images.
10. After ensuring gallery images display ok, save the post and confirm for the frontend
11. Try styling the image block via theme.json (see snippet below)
12. Switch to the site editor and navigate within the Global Styles sidebar to Blocks > Image > Layout and test configuring borders on images.
12. Run `npm run test-unit` and ensure the image fixture tests pass

```
{
	"styles": {
		"core/image": {
			"border": {
				"width": "15px",
				"style": "dashed",
				"color": "blue"
			}
		}
	}
}
```

## Screenshots

https://user-images.githubusercontent.com/60436221/179129538-44627ad9-e892-4325-85db-45563e4a4916.mp4


## Types of changes
New feature.

## Checklist:
